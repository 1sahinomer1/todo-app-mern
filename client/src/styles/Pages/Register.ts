import styled from 'styled-components';

import { colors } from 'theme';

export const Redirect = styled.div`
  margin-top: 20px;
  display: flex;
  justify-content: center;
`;
export const LinkText = styled.p`
  color: ${colors.primaryColor};
  cursor: pointer;
  font-weight: bold;
`;
export const Form = styled.form`
  width: 30%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 50px;
`;
