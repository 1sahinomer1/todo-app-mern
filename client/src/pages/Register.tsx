import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { Link, useHistory } from 'react-router-dom';
import { toast } from 'react-toastify';
import * as yup from 'yup';

import { Button, Input } from 'components';

import LoginRegister from 'layouts/LoginRegister/LoginRegister';
import { formTypes } from 'constants/layouts/loginRegister';
import { formValidations } from 'utils';

import * as S from 'styles/Pages/Register';
import axios from 'axios';

interface IFormInputs {
  fullname: string;
  email: string;
  password: string;
  passwordConfirmation: string;
}

const Register = () => {
  const history = useHistory();

  const validationSchema = yup
    .object()
    .shape({
      fullname: formValidations.fullNameValidation,
      email: formValidations.emailValidation,
      password: formValidations.passwordValidation,
      passwordConfirmation: formValidations.passwordConfirmation,
    })
    .required();

  const {
    register,
    handleSubmit,
    watch,
    formState: { errors },
  } = useForm<IFormInputs>({
    resolver: yupResolver(validationSchema),
  });
  const onSubmit = async (data: IFormInputs) => {
    const User = {
      name: data.fullname,
      email: data.email,
      password: data.password,
    };
    try {
      const response = await axios.post('/users/register', User);
      setTimeout(() => {
        history.push('/login');
      }, 3000);
      toast.success(response.data.message);
    } catch (error) {
      toast.error((error as any).response.data.message);
    }
  };

  return (
    <LoginRegister formType={formTypes.REGISTER}>
      <S.Form onSubmit={handleSubmit(onSubmit)}>
        <Input
          {...register('fullname')}
          type="text"
          placeholder="Full name"
          error={errors.fullname?.message}
          label="Fullname"
          hasValue={watch('fullname')}
        />

        <Input
          {...register('email')}
          type="text"
          label="Email"
          placeholder="Email"
          error={errors.email?.message}
          hasValue={watch('email')}
        />
        <Input
          {...register('password')}
          type="password"
          label="Password"
          placeholder="Password"
          error={errors.password?.message}
          hasValue={watch('password')}
        />
        <Input
          {...register('passwordConfirmation')}
          type="password"
          label="Password Again"
          placeholder="Password Again"
          error={errors.passwordConfirmation?.message}
          hasValue={watch('passwordConfirmation')}
        />
        <Button data-testid="registerBtn">Register</Button>

        <S.Redirect>
          Already have Account ?
          <Link to="/login">
            <S.LinkText data-testid="signInLink">Sign In</S.LinkText>
          </Link>
        </S.Redirect>
      </S.Form>
    </LoginRegister>
  );
};

export default Register;
