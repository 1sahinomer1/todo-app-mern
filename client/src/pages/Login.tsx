import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { Link, useHistory } from 'react-router-dom';
import * as yup from 'yup';
import axios from 'axios';
import { toast } from 'react-toastify';
import { useCookies } from 'react-cookie';

import { Button, Input } from 'components';

import { formTypes } from 'constants/layouts/loginRegister';
import LoginRegister from 'layouts/LoginRegister/LoginRegister';
import { formValidations } from 'utils';

import * as S from 'styles/Pages/Register';

const Login = () => {
  const history = useHistory();
  const [cookies, setCookie] = useCookies(['users']);

  interface IFormInputs {
    fullname: string;
    email: string;
    password: string;
    passwordConfirmation: string;
  }
  const validationSchema = yup
    .object()
    .shape({
      email: formValidations.emailValidation,
      password: formValidations.passwordValidation,
    })
    .required();

  const {
    register,
    handleSubmit,
    watch,
    formState: { errors },
  } = useForm<IFormInputs>({
    resolver: yupResolver(validationSchema),
  });
  const onSubmit = async (data: IFormInputs) => {
    const User = {
      email: data.email,
      password: data.password,
    };
    try {
      const response = await axios.post('/users/login', User);
      setTimeout(() => {
        history.push('/');
      }, 3000);
      toast.success(response.data.message);
      setCookie('users', response.data, { path: '/' });
    } catch (error) {
      toast.error((error as any).response.data.message);
    }
  };
  return (
    <LoginRegister formType={formTypes.LOGIN}>
      <S.Form onSubmit={handleSubmit(onSubmit)}>
        <Input
          {...register('email')}
          type="text"
          label="Email"
          placeholder="Email"
          error={errors.email?.message}
          hasValue={watch('email')}
        />
        <Input
          {...register('password')}
          type="password"
          label="Password"
          placeholder="Password"
          error={errors.password?.message}
          hasValue={watch('password')}
        />

        <Button data-testid="loginBtn">Login</Button>
        <S.Redirect>
          Don't you have an account ?
          <Link to="/register">
            <S.LinkText data-testid="signupLink">Sign Up</S.LinkText>
          </Link>
        </S.Redirect>
      </S.Form>
    </LoginRegister>
  );
};

export default Login;
