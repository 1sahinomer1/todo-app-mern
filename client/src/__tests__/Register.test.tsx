import { render, screen, waitFor, act } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';

import Login from 'pages/Login';
import Register from 'pages/Register';

import { Router } from 'react-router';

test('SignIn snapshot', async () => {
  const history = createMemoryHistory();

  const pages = render(
    <Router history={history}>
      <Login />
      <Register />
    </Router>
  );
  await waitFor(() => pages);
  expect(pages).toMatchSnapshot();
});

test('SignIn page loaded properly', () => {
  const history = createMemoryHistory();

  render(
    <Router history={history}>
      <Login />
      <Register />
    </Router>
  );
  expect(screen.getByText(/register account/i)).toBeInTheDocument();
  const signUpLink = screen.getByTestId('signInLink');
  userEvent.click(signUpLink);
  expect(screen.getByText(/welcome todoapp/i)).toBeInTheDocument();
});

test('Trying to register with unfilled form', async () => {
  //When I clicked the button,,
  // it gave an act error and when wrapped in this way, the error disappeared.
  await act(async () => {
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <Register />
      </Router>
    );
    const registerButton = screen.getByRole('button');
    await waitFor(() => {
      userEvent.click(registerButton);
    });

    screen.debug();
    // expect(
    //   screen.getByText('Full name is a required field')
    // ).toBeInTheDocument();
    // expect(screen.getByText('Email is a required field')).toBeInTheDocument();
    // expect(
    //   screen.getByText('Password is a required field')
    // ).toBeInTheDocument();
    // expect(
    //   screen.getByText('Confirm Password is required')
    // ).toBeInTheDocument();
  });
});
