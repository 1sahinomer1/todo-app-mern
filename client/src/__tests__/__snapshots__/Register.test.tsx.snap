// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jrQzAO kURLVg"
      >
        <div
          class="sc-kDTinF djgIZb"
        >
          <h1
            class="sc-iqseJM TPmhY"
          >
            Welcome TodoApp
          </h1>
          <p
            class="sc-crHmcD WZwVv"
          >
            Sign in to create your to-do list
          </p>
        </div>
        <form
          class="sc-ksdxgE cuPkws"
        >
          <div
            class="sc-dkPtRN UxpTq"
          >
            <h3
              class="sc-furwcr fuauLV"
            >
              Email
            </h3>
            <div
              class="sc-hKwDye fSawMy"
            >
              <input
                class="sc-gKclnd hCRLHi"
                data-testid="input"
                name="email"
                placeholder="Email"
                type="text"
              />
            </div>
            <p
              class="sc-iCfMLu gNrjYY"
              data-testid="errorMessage"
            />
          </div>
          <div
            class="sc-dkPtRN UxpTq"
          >
            <h3
              class="sc-furwcr fuauLV"
            >
              Password
            </h3>
            <div
              class="sc-hKwDye fSawMy"
            >
              <input
                class="sc-gKclnd hCRLHi"
                data-testid="input"
                name="password"
                placeholder="Password"
                type="password"
              />
              <svg
                class="sc-eCImPb dmhoxo"
                data-testid="eye"
                fill="none"
                height="20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#prefix__clip0_16:2527)"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10 2.212c5.16 0 8.1 3.532 9.392 5.637a4.09 4.09 0 010 4.302C18.1 14.256 15.16 17.788 10 17.788c-5.16 0-8.1-3.532-9.392-5.637a4.09 4.09 0 010-4.302C1.9 5.744 4.84 2.212 10 2.212zm0 13.909c4.35 0 6.862-3.038 7.972-4.843a2.431 2.431 0 000-2.556C16.862 6.913 14.349 3.879 10 3.879c-4.35 0-6.862 3.038-7.972 4.843a2.432 2.432 0 000 2.556c1.11 1.805 3.623 4.843 7.972 4.843zM7.685 6.536a4.167 4.167 0 114.63 6.928 4.167 4.167 0 01-4.63-6.928zm.926 5.543A2.5 2.5 0 1011.39 7.92 2.5 2.5 0 008.61 12.08z"
                    fill="#50C2C9"
                    fill-rule="evenodd"
                  />
                </g>
                <defs>
                  <clippath
                    id="prefix__clip0_16:2527"
                  >
                    <path
                      d="M0 0h20v20H0z"
                      fill="#fff"
                    />
                  </clippath>
                </defs>
              </svg>
            </div>
            <p
              class="sc-iCfMLu gNrjYY"
              data-testid="errorMessage"
            />
          </div>
          <button
            class="sc-pVTFL dWUFRl"
            data-testid="loginBtn"
          >
            Login
          </button>
          <div
            class="sc-egiyK GqKQt"
          >
            Don't you have an account ?
            <a
              href="/register"
            >
              <p
                class="sc-bqiRlB eioUKt"
                data-testid="signupLink"
              >
                Sign Up
              </p>
            </a>
          </div>
        </form>
        <div
          class="Toastify"
        />
      </div>
      <div
        class="sc-jrQzAO kURLVg"
      >
        <div
          class="sc-kDTinF djgIZb"
        >
          <h1
            class="sc-iqseJM TPmhY"
          >
            Register Account
          </h1>
          <p
            class="sc-crHmcD WZwVv"
          >
            Sign up to create your to-do list
          </p>
        </div>
        <form
          class="sc-ksdxgE cuPkws"
        >
          <div
            class="sc-dkPtRN UxpTq"
          >
            <h3
              class="sc-furwcr fuauLV"
            >
              Fullname
            </h3>
            <div
              class="sc-hKwDye fSawMy"
            >
              <input
                class="sc-gKclnd hCRLHi"
                data-testid="input"
                name="fullname"
                placeholder="Full name"
                type="text"
              />
            </div>
            <p
              class="sc-iCfMLu gNrjYY"
              data-testid="errorMessage"
            />
          </div>
          <div
            class="sc-dkPtRN UxpTq"
          >
            <h3
              class="sc-furwcr fuauLV"
            >
              Email
            </h3>
            <div
              class="sc-hKwDye fSawMy"
            >
              <input
                class="sc-gKclnd hCRLHi"
                data-testid="input"
                name="email"
                placeholder="Email"
                type="text"
              />
            </div>
            <p
              class="sc-iCfMLu gNrjYY"
              data-testid="errorMessage"
            />
          </div>
          <div
            class="sc-dkPtRN UxpTq"
          >
            <h3
              class="sc-furwcr fuauLV"
            >
              Password
            </h3>
            <div
              class="sc-hKwDye fSawMy"
            >
              <input
                class="sc-gKclnd hCRLHi"
                data-testid="input"
                name="password"
                placeholder="Password"
                type="password"
              />
              <svg
                class="sc-eCImPb dmhoxo"
                data-testid="eye"
                fill="none"
                height="20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#prefix__clip0_16:2527)"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10 2.212c5.16 0 8.1 3.532 9.392 5.637a4.09 4.09 0 010 4.302C18.1 14.256 15.16 17.788 10 17.788c-5.16 0-8.1-3.532-9.392-5.637a4.09 4.09 0 010-4.302C1.9 5.744 4.84 2.212 10 2.212zm0 13.909c4.35 0 6.862-3.038 7.972-4.843a2.431 2.431 0 000-2.556C16.862 6.913 14.349 3.879 10 3.879c-4.35 0-6.862 3.038-7.972 4.843a2.432 2.432 0 000 2.556c1.11 1.805 3.623 4.843 7.972 4.843zM7.685 6.536a4.167 4.167 0 114.63 6.928 4.167 4.167 0 01-4.63-6.928zm.926 5.543A2.5 2.5 0 1011.39 7.92 2.5 2.5 0 008.61 12.08z"
                    fill="#50C2C9"
                    fill-rule="evenodd"
                  />
                </g>
                <defs>
                  <clippath
                    id="prefix__clip0_16:2527"
                  >
                    <path
                      d="M0 0h20v20H0z"
                      fill="#fff"
                    />
                  </clippath>
                </defs>
              </svg>
            </div>
            <p
              class="sc-iCfMLu gNrjYY"
              data-testid="errorMessage"
            />
          </div>
          <div
            class="sc-dkPtRN UxpTq"
          >
            <h3
              class="sc-furwcr fuauLV"
            >
              Password Again
            </h3>
            <div
              class="sc-hKwDye fSawMy"
            >
              <input
                class="sc-gKclnd hCRLHi"
                data-testid="input"
                name="passwordConfirmation"
                placeholder="Password Again"
                type="password"
              />
              <svg
                class="sc-eCImPb dmhoxo"
                data-testid="eye"
                fill="none"
                height="20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#prefix__clip0_16:2527)"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10 2.212c5.16 0 8.1 3.532 9.392 5.637a4.09 4.09 0 010 4.302C18.1 14.256 15.16 17.788 10 17.788c-5.16 0-8.1-3.532-9.392-5.637a4.09 4.09 0 010-4.302C1.9 5.744 4.84 2.212 10 2.212zm0 13.909c4.35 0 6.862-3.038 7.972-4.843a2.431 2.431 0 000-2.556C16.862 6.913 14.349 3.879 10 3.879c-4.35 0-6.862 3.038-7.972 4.843a2.432 2.432 0 000 2.556c1.11 1.805 3.623 4.843 7.972 4.843zM7.685 6.536a4.167 4.167 0 114.63 6.928 4.167 4.167 0 01-4.63-6.928zm.926 5.543A2.5 2.5 0 1011.39 7.92 2.5 2.5 0 008.61 12.08z"
                    fill="#50C2C9"
                    fill-rule="evenodd"
                  />
                </g>
                <defs>
                  <clippath
                    id="prefix__clip0_16:2527"
                  >
                    <path
                      d="M0 0h20v20H0z"
                      fill="#fff"
                    />
                  </clippath>
                </defs>
              </svg>
            </div>
            <p
              class="sc-iCfMLu gNrjYY"
              data-testid="errorMessage"
            />
          </div>
          <button
            class="sc-pVTFL dWUFRl"
            data-testid="registerBtn"
          >
            Register
          </button>
          <div
            class="sc-egiyK GqKQt"
          >
            Already have Account ?
            <a
              href="/login"
            >
              <p
                class="sc-bqiRlB eioUKt"
                data-testid="signInLink"
              >
                Sign In
              </p>
            </a>
          </div>
        </form>
        <div
          class="Toastify"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jrQzAO kURLVg"
    >
      <div
        class="sc-kDTinF djgIZb"
      >
        <h1
          class="sc-iqseJM TPmhY"
        >
          Welcome TodoApp
        </h1>
        <p
          class="sc-crHmcD WZwVv"
        >
          Sign in to create your to-do list
        </p>
      </div>
      <form
        class="sc-ksdxgE cuPkws"
      >
        <div
          class="sc-dkPtRN UxpTq"
        >
          <h3
            class="sc-furwcr fuauLV"
          >
            Email
          </h3>
          <div
            class="sc-hKwDye fSawMy"
          >
            <input
              class="sc-gKclnd hCRLHi"
              data-testid="input"
              name="email"
              placeholder="Email"
              type="text"
            />
          </div>
          <p
            class="sc-iCfMLu gNrjYY"
            data-testid="errorMessage"
          />
        </div>
        <div
          class="sc-dkPtRN UxpTq"
        >
          <h3
            class="sc-furwcr fuauLV"
          >
            Password
          </h3>
          <div
            class="sc-hKwDye fSawMy"
          >
            <input
              class="sc-gKclnd hCRLHi"
              data-testid="input"
              name="password"
              placeholder="Password"
              type="password"
            />
            <svg
              class="sc-eCImPb dmhoxo"
              data-testid="eye"
              fill="none"
              height="20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#prefix__clip0_16:2527)"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 2.212c5.16 0 8.1 3.532 9.392 5.637a4.09 4.09 0 010 4.302C18.1 14.256 15.16 17.788 10 17.788c-5.16 0-8.1-3.532-9.392-5.637a4.09 4.09 0 010-4.302C1.9 5.744 4.84 2.212 10 2.212zm0 13.909c4.35 0 6.862-3.038 7.972-4.843a2.431 2.431 0 000-2.556C16.862 6.913 14.349 3.879 10 3.879c-4.35 0-6.862 3.038-7.972 4.843a2.432 2.432 0 000 2.556c1.11 1.805 3.623 4.843 7.972 4.843zM7.685 6.536a4.167 4.167 0 114.63 6.928 4.167 4.167 0 01-4.63-6.928zm.926 5.543A2.5 2.5 0 1011.39 7.92 2.5 2.5 0 008.61 12.08z"
                  fill="#50C2C9"
                  fill-rule="evenodd"
                />
              </g>
              <defs>
                <clippath
                  id="prefix__clip0_16:2527"
                >
                  <path
                    d="M0 0h20v20H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="sc-iCfMLu gNrjYY"
            data-testid="errorMessage"
          />
        </div>
        <button
          class="sc-pVTFL dWUFRl"
          data-testid="loginBtn"
        >
          Login
        </button>
        <div
          class="sc-egiyK GqKQt"
        >
          Don't you have an account ?
          <a
            href="/register"
          >
            <p
              class="sc-bqiRlB eioUKt"
              data-testid="signupLink"
            >
              Sign Up
            </p>
          </a>
        </div>
      </form>
      <div
        class="Toastify"
      />
    </div>
    <div
      class="sc-jrQzAO kURLVg"
    >
      <div
        class="sc-kDTinF djgIZb"
      >
        <h1
          class="sc-iqseJM TPmhY"
        >
          Register Account
        </h1>
        <p
          class="sc-crHmcD WZwVv"
        >
          Sign up to create your to-do list
        </p>
      </div>
      <form
        class="sc-ksdxgE cuPkws"
      >
        <div
          class="sc-dkPtRN UxpTq"
        >
          <h3
            class="sc-furwcr fuauLV"
          >
            Fullname
          </h3>
          <div
            class="sc-hKwDye fSawMy"
          >
            <input
              class="sc-gKclnd hCRLHi"
              data-testid="input"
              name="fullname"
              placeholder="Full name"
              type="text"
            />
          </div>
          <p
            class="sc-iCfMLu gNrjYY"
            data-testid="errorMessage"
          />
        </div>
        <div
          class="sc-dkPtRN UxpTq"
        >
          <h3
            class="sc-furwcr fuauLV"
          >
            Email
          </h3>
          <div
            class="sc-hKwDye fSawMy"
          >
            <input
              class="sc-gKclnd hCRLHi"
              data-testid="input"
              name="email"
              placeholder="Email"
              type="text"
            />
          </div>
          <p
            class="sc-iCfMLu gNrjYY"
            data-testid="errorMessage"
          />
        </div>
        <div
          class="sc-dkPtRN UxpTq"
        >
          <h3
            class="sc-furwcr fuauLV"
          >
            Password
          </h3>
          <div
            class="sc-hKwDye fSawMy"
          >
            <input
              class="sc-gKclnd hCRLHi"
              data-testid="input"
              name="password"
              placeholder="Password"
              type="password"
            />
            <svg
              class="sc-eCImPb dmhoxo"
              data-testid="eye"
              fill="none"
              height="20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#prefix__clip0_16:2527)"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 2.212c5.16 0 8.1 3.532 9.392 5.637a4.09 4.09 0 010 4.302C18.1 14.256 15.16 17.788 10 17.788c-5.16 0-8.1-3.532-9.392-5.637a4.09 4.09 0 010-4.302C1.9 5.744 4.84 2.212 10 2.212zm0 13.909c4.35 0 6.862-3.038 7.972-4.843a2.431 2.431 0 000-2.556C16.862 6.913 14.349 3.879 10 3.879c-4.35 0-6.862 3.038-7.972 4.843a2.432 2.432 0 000 2.556c1.11 1.805 3.623 4.843 7.972 4.843zM7.685 6.536a4.167 4.167 0 114.63 6.928 4.167 4.167 0 01-4.63-6.928zm.926 5.543A2.5 2.5 0 1011.39 7.92 2.5 2.5 0 008.61 12.08z"
                  fill="#50C2C9"
                  fill-rule="evenodd"
                />
              </g>
              <defs>
                <clippath
                  id="prefix__clip0_16:2527"
                >
                  <path
                    d="M0 0h20v20H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="sc-iCfMLu gNrjYY"
            data-testid="errorMessage"
          />
        </div>
        <div
          class="sc-dkPtRN UxpTq"
        >
          <h3
            class="sc-furwcr fuauLV"
          >
            Password Again
          </h3>
          <div
            class="sc-hKwDye fSawMy"
          >
            <input
              class="sc-gKclnd hCRLHi"
              data-testid="input"
              name="passwordConfirmation"
              placeholder="Password Again"
              type="password"
            />
            <svg
              class="sc-eCImPb dmhoxo"
              data-testid="eye"
              fill="none"
              height="20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#prefix__clip0_16:2527)"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 2.212c5.16 0 8.1 3.532 9.392 5.637a4.09 4.09 0 010 4.302C18.1 14.256 15.16 17.788 10 17.788c-5.16 0-8.1-3.532-9.392-5.637a4.09 4.09 0 010-4.302C1.9 5.744 4.84 2.212 10 2.212zm0 13.909c4.35 0 6.862-3.038 7.972-4.843a2.431 2.431 0 000-2.556C16.862 6.913 14.349 3.879 10 3.879c-4.35 0-6.862 3.038-7.972 4.843a2.432 2.432 0 000 2.556c1.11 1.805 3.623 4.843 7.972 4.843zM7.685 6.536a4.167 4.167 0 114.63 6.928 4.167 4.167 0 01-4.63-6.928zm.926 5.543A2.5 2.5 0 1011.39 7.92 2.5 2.5 0 008.61 12.08z"
                  fill="#50C2C9"
                  fill-rule="evenodd"
                />
              </g>
              <defs>
                <clippath
                  id="prefix__clip0_16:2527"
                >
                  <path
                    d="M0 0h20v20H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="sc-iCfMLu gNrjYY"
            data-testid="errorMessage"
          />
        </div>
        <button
          class="sc-pVTFL dWUFRl"
          data-testid="registerBtn"
        >
          Register
        </button>
        <div
          class="sc-egiyK GqKQt"
        >
          Already have Account ?
          <a
            href="/login"
          >
            <p
              class="sc-bqiRlB eioUKt"
              data-testid="signInLink"
            >
              Sign In
            </p>
          </a>
        </div>
      </form>
      <div
        class="Toastify"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
